### DISCUSSION TITLE
What are the restrictions on the table name?

### DISCUSSION THREAD

**Discussion by yxmm-wxe** [duckdb/duckdb#2160] [Q&A] [answered]
In my test, the table name is more than 50,000 in length, including English, Chinese, Polish, numbers, and special characters. 
In this case, it can still be created successfully as long as it is enclosed in double quotation marks.

So I want to know what are the restrictions on the table name?  like length, special characters that can be included, etc.
I cannot find the instructions in the documents.

**Comment by Mytherin** [comment, marked as answer]
Currently there are no restrictions on the table name except that it cannot be empty. Perhaps we should add an artificial length restriction to the table name and column names, as I can't imagine table names of e.g. >65K bytes would be intentional and they would bloat the storage.

**Reply by yxmm-wxe** [reply]
I think it might be a good thing. Duckdb is an in-memory database and column and table names should be managed by the user. If the user's memory is large enough, it doesn't matter if the table name is longer.
